src/mlmc/tool/base_process.py

```diff
@@ -0,0 +1,392 @@
+        self.append = False
+        self.clean = args.clean
+
+        if args.command == 'run':
+            self.run()
+        else:
+            self.append = True
+            self.clean = False
+            self.run(renew=True) if args.command == 'renew' else self.run()
+
+    def get_arguments(self, arguments):
+        """
+        Getting arguments from console
+        :param arguments: list of arguments
+        :return: namespace
+        """
+        import argparse
+        parser = argparse.ArgumentParser()
+
+        parser.add_argument('command', choices=['run', 'collect', 'renew'],
+                            help='run - create new execution,'
+                                 'collect - keep collected, append existing HDF file'
+                                 'renew - renew failed samples, run new samples with failed sample ids (which determine random seed)')
+        parser.add_argument('work_dir', help='Work directory')
+        parser.add_argument("-c", "--clean", default=False, action='store_true',
+                            help="Clean before run, used only with 'run' command")
+
+        args = parser.parse_args(arguments)
+
+        return args
+
+    def run(self, renew=True):
+        """
+        Run mlmc
+        :return: None
+        """
+        os.makedirs(self.work_dir, mode=0o775, exist_ok=True)
+
+        mlmc_list = []
+        for nl in [1]:  # , 2, 3, 4,5, 7, 9]:
+            mlmc = self.setup_config(nl, clean=True)
+            self.generate_jobs(mlmc, n_samples=[8], sample_sleep=self.sample_sleep, sample_timeout=self.sample_timeout)
+            mlmc_list.append(mlmc)
+
+        self.all_collect(mlmc_list)
+
+    # def collect(self):
+    #     """
+    #     Collect samples
+    #     :return: None
+    #     """
+    #     assert os.path.isdir(self.work_dir)
+    #     mlmc_list = []
+    #
+    #     for nl in [1, 2, 3, 4, 5, 7]:  # , 3, 4, 5, 7, 9]:#, 5,7]:
+    #         mlmc = self.setup_config(nl, clean=False)
+    #         mlmc_list.append(mlmc)
+    #     self.all_collect(mlmc_list)
+    #     self.calculate_var(mlmc_list)
+    #     # show_results(mlmc_list)
+
+    # def process(self):
+    #     """
+    #     Use collected data
+    #     :return: None
+    #     """
+    #     assert os.path.isdir(self.work_dir)
+    #     mlmc_est_list = []
+    #     # for nl in [ 1,3,5,7,9]:
+    #     for nl in [3]:  # high resolution fields
+    #         mlmc = self.setup_config(nl, clean=False)
+    #         # Use wrapper object for working with collected data
+    #         mlmc_est_list.append(mlmc)
+    #
+    #     cl = CompareLevels(mlmc_est_list,
+    #                        output_dir=src_path,
+    #                        quantity_name="Q [m/s]",
+    #                        moment_class=Legendre,
+    #                        log_scale=False,
+    #                        n_moments=21, )
+    #
+    #     self.process_analysis(cl)
+
+    def set_environment_variables(self):
```

Comment:
Separate finalized and temporary solution. E.g. various analysis and plotting functions should not be part of ProcessBase. 

The purpose of ProcessBase is to process common script arguments and make a backbone of the sampling script.